proofs/refine.ml

```diff
@@ -93,17 +91,17 @@ let generic_refine ~typecheck f gl =
         | Some id -> Evd.rename evk id sigma
   in
   (* Mark goals *)
-  let sigma = Proofview.Unsafe.mark_as_goals sigma comb in
-  let sigma = Proofview.Unsafe.mark_unresolvables sigma shelf in
-  let comb = CList.map (fun x -> Proofview.goal_with_state x state) comb in
+  let sigma = Proofview.Unsafe.mark_as_goals sigma future_goals.Evd.future_comb in
+  (* FIXME
+  let sigma = Proofview.Unsafe.mark_unresolvables sigma future_goals.Evd.future_shelf in
```

Comment:
@mattam82 commenting out this line didn't seem to have any impact. Would have an example showing why it is needed?

Reply:
It's used to disallow typeclass resolution on shelved goals, I think you can find examples where it has an impact in Arthur Charguéraud's  issues  related to apply/eapply

Reply:
Yeah, I saw the discussion and tried to extract an example but failed. I was also surprised the test suite does not fail.

Reply:
This marking does not seems necessary anymore (at least, I couldn't observe any change, including on Arthurs's examples), so I removed it.